app/(main)/details/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback } from "react";
+import { Image, Linking, ScrollView, Text, View } from "react-native";
+import { useStyles } from "react-native-unistyles";
+import { useTranslation } from "react-i18next";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+import Toolbar from "@/components/toolbar";
+import { ArticleType } from "@/types/home";
+import { useLocalSearchParams } from "expo-router";
+
+const DetailsPage: React.FC = () => {
+  const { d } = useLocalSearchParams();
+  const article: ArticleType = JSON.parse(d as string);
+  const { t } = useTranslation();
+  const { top, bottom } = useSafeAreaInsets();
+  const { styles } = useStyles((theme: any) => ({
+    container: {
+      flex: 1,
+      backgroundColor: theme.colors.background,
+      paddingBottom: bottom,
+    },
+    toolbar: {
+      flexDirection: "row",
+      justifyContent: "space-between",
+      alignItems: "center",
+      padding: theme.spacing.md,
+      backgroundColor: theme.colors.primary,
+      paddingTop: top + theme.spacing.md,
+    },
+    toolbarTitle: {
+      color: "white",
+      fontSize: 18,
+      fontWeight: "bold",
+    },
+
+    scrollContainer: {
+      padding: 16,
+      backgroundColor: "#fff",
+      minHeight: "95%",
+    },
+    image: {
+      width: "100%",
+      height: 200,
+      marginBottom: 16,
+      borderRadius: 8,
+    },
+    title: {
+      fontSize: 22,
+      fontWeight: "bold",
+      marginBottom: 8,
+    },
+    author: {
+      fontSize: 14,
+      color: "#555",
+      marginBottom: 8,
+    },
+    description: {
+      fontSize: 16,
+      color: "#333",
+      marginBottom: 12,
+    },
+    content: {
+      fontSize: 14,
+      color: "#333",
+      lineHeight: 22,
+      marginBottom: 16,
+    },
+    source: {
+      fontSize: 12,
+      color: "#777",
+      marginBottom: 16,
+    },
+    link: {
+      fontSize: 16,
+      color: "#0066cc",
+      textDecorationLine: "underline",
+    },
+  }));
```

Comment:
following unistyles [docs](https://v2.unistyl.es/start/introduction/), this needs to be split in two spots:

```tsx
const { styles } = useStyles(stylesheet) // in component
...
const stylesheet = createStyleSheet(theme => ({ // outside of component
  ...
})
```

we shouldn't also use the type `any`, in strict TS im against using `any` or even `unknown`. if you're getting a TS error its because unistyles isn't fully set up right